crypto/x509/x509_vfy.c

```diff
@@ -571,6 +572,9 @@ static int check_chain_extensions(X509_STORE_CTX *ctx)
             /* Check sig alg consistency acc. to RFC 5280 section 4.1.1.2 */
             if (X509_ALGOR_cmp(&x->sig_alg, &x->cert_info.signature) != 0)
                 ctx->error = X509_V_ERR_SIGNATURE_ALGORITHM_INCONSISTENCY;
+            /* Check for presence of explicit elliptic curve parameters */
+            if (!check_curve(x))
+                ctx->error = X509_V_ERR_CA_KEY_EXPLICIT_PARAMS;
```

Comment:
I'd say your new check is well placed here,
yet I just realized the name of the function `check_chain_extensions()` is not really adequate any more -
it meanwhile does checks not only regarding X509v3 extensions.
Would you mind generalizing the name of this (internal) function?

Reply:
Any proposal for the new name? Simply check_chain?

Reply:
I've decided to use check_chain.